Client/Frontend/Browser/TabManager.swift

```diff
@@ -230,6 +229,10 @@ class TabManager : NSObject {
         }
         assert(count == prevCount - 1, "Tab removed")
 
+        if tab != selectedTab {
+            _selectedIndex = selectedTab == nil ? 0 : find(tabs, selectedTab!) ?? 0
```

Comment:
For the `selectedTab == nil` case, I think we want to return -1 instead of 0. [We start with -1 as unselected](https://github.com/mozilla/firefox-ios/blob/d5c27808a685685fa7826fc941125603b4a1b024/Client/Frontend/Browser/TabManager.swift#L53) by default, and using -1 means [we'll be outside tab range](https://github.com/mozilla/firefox-ios/blob/d5c27808a685685fa7826fc941125603b4a1b024/Client/Frontend/Browser/TabManager.swift#L89) so we can return nil.